test/metabase/test/data/sql.clj

```diff
@@ -198,38 +198,43 @@
   tx/dispatch-on-driver-with-test-extensions
   :hierarchy #'driver/hierarchy)
 
+(defn- format-and-quote-field-name [driver field-name]
+  (sql.u/quote-name driver :field (ddl.i/format-name driver field-name)))
+
+(defn- field-definition-sql
+  [driver {:keys [field-name base-type field-comment not-null?], :as field-definition}]
+  (let [field-name (format-and-quote-field-name driver field-name)
+        field-type (or (cond
+                         (and (map? base-type) (contains? base-type :native))
+                         (:native base-type)
+
+                         (and (map? base-type) (contains? base-type :natives))
+                         (get-in base-type [:natives driver])
+
+                         base-type
+                         (field-base-type->sql-type driver base-type))
+                       (throw (ex-info (format "Missing datatype for field %s for driver: %s"
+                                               field-name driver)
+                                       {:field  field-definition
+                                        :driver driver})))
+        not-null       (when not-null?
+                         "NOT NULL")
+        inline-comment (inline-column-comment-sql driver field-comment)]
+    (str/join " " (filter some? [field-name field-type not-null inline-comment]))))
```

Comment:
I refactored this a bit and moved the code for field definition DDS into its own function. The `create-table-sql` method was kind of big at 30 lines; moving half of that out into a separate function makes both easier to read.

I also refactored this stuff a bit to make the way the string gets built from different parts more obvious.

I also added support for `NOT NULL` as mentioned earlier, needed to be able to easily test failing inserts